bin/improver-spot-extract

```diff
@@ -187,15 +193,81 @@ def main():
         site_properties = site_dictionary(args.latitudes, args.longitudes,
                                           args.altitudes)
 
-    run_spotdata(args.config_file_path, args.data_path, args.ancillary_path,
-                 diagnostic_list=args.diagnostics, site_path=args.site_path,
-                 constants_file_path=args.constants_path,
-                 site_properties=site_properties,
-                 forecast_date=args.forecast_date,
-                 forecast_time=args.forecast_time,
-                 forecast_length=args.forecast_length,
-                 output_path=args.output_path,
-                 use_multiprocessing=args.multiprocess)
+    # Check site data has been provided.
+    if args.site_path is None and not site_properties:
+        raise ValueError("No SpotData site information has been provided "
+                         "from a file or defined at runtime.")
+
+    # If using locations set at command line, set optional information such
+    # as site altitude and site_id. If a site definition file is provided it
+    # will take precedence.
+    if args.site_path is None:
+        sites = ImportSiteData('runtime_list').process(site_properties)
+    else:
+        sites = ImportSiteData('from_file').process(args.site_path)
+
+    # Read in extraction recipes for all diagnostics.
+    with open(args.config_file_path, 'r') as input_file:
+        diagnostics = json.load(input_file)
```

Comment:
This file is now treated as the list of diagnostics I want to process, rather than just a look up for how to process any given diagnostic that the user might select (see comment below).

Reply:
Edited to make use of the `diagnostics` optional argument.